radix-engine-interface/src/model/resource/non_fungible_global_id.rs

```diff
@@ -204,56 +159,51 @@ mod tests {
         assert_eq!(
             NonFungibleGlobalId::try_from_canonical_string(
                 &dec,
-                NonFungibleIdType::Number,
-                "resource_sim1qzntya3nlyju8zsj8h86fz8ma5yl8smwjlg9tckkqvrs520k2p:1",
+                "resource_sim1qzntya3nlyju8zsj8h86fz8ma5yl8smwjlg9tckkqvrs520k2p:<id>",
             )
             .unwrap()
             .to_canonical_string(&enc),
-            "resource_sim1qzntya3nlyju8zsj8h86fz8ma5yl8smwjlg9tckkqvrs520k2p:1"
+            "resource_sim1qzntya3nlyju8zsj8h86fz8ma5yl8smwjlg9tckkqvrs520k2p:<id>"
         );
 
         assert_eq!(
             NonFungibleGlobalId::try_from_canonical_string(
                 &dec,
-                NonFungibleIdType::Number,
-                "resource_sim1qzntya3nlyju8zsj8h86fz8ma5yl8smwjlg9tckkqvrs520k2p:10",
+                "resource_sim1qzntya3nlyju8zsj8h86fz8ma5yl8smwjlg9tckkqvrs520k2p:#123#",
             )
             .unwrap()
             .to_canonical_string(&enc),
-            "resource_sim1qzntya3nlyju8zsj8h86fz8ma5yl8smwjlg9tckkqvrs520k2p:10"
+            "resource_sim1qzntya3nlyju8zsj8h86fz8ma5yl8smwjlg9tckkqvrs520k2p:#123#"
         );
 
         assert_eq!(
             NonFungibleGlobalId::try_from_canonical_string(
                 &dec,
-                NonFungibleIdType::UUID,
-                "resource_sim1qzntya3nlyju8zsj8h86fz8ma5yl8smwjlg9tckkqvrs520k2p:1234567890",
+                "resource_sim1qzntya3nlyju8zsj8h86fz8ma5yl8smwjlg9tckkqvrs520k2p:{1234567890}",
```

Comment:
I don't think we should support such non-conformant IDs

Reply:
Initially thought the same way, but found `uuid` (crate) is taking a mroe lenient approach, https://docs.rs/uuid/1.2.2/uuid/struct.Uuid.html#method.parse_str
> Any of the formats generated by this module (simple, hyphenated, urn, Microsoft GUID) are supported by this parsing function.

Happy to change it to allow braced form only.

Reply:
I feel it'd probably be better to be stricter, ideally so there's a unique representation.

We may even want to enforce eg lower case - what do you think?

Reply:
[RFC 4122 Section 3](https://tools.ietf.org/html/rfc4122#section-3) requires that the characters be generated in lower case, while being case-insensitive on input.

I figure we'd limit the format but be case-insensitive, for the parsing part.

Reply:
Sounds reasonable to me 👍

Reply:
Strengthened validation for textual representation.